### ISSUE TITLE
batch size = 1

### ISSUE THREAD

**Issue by hdjsjyl** [raywang4/DispLoss#2]
Hi Author,
     I hope to apply DispLoss to Unet-based diffusion model.
     1. The batch size ==1, the loss is constantly 0, is there any wrong here? If not, batch size =1 is it meaningful to apply disploss?
     2. For which feature maps of blocks in unet-based diffusion model, is there any suggestions here?

Thank you.

**Comment by raywang4** [comment]
1. Similar to Contrastive Loss, Dispersive Loss requires more than one sample in a batch to effectively function. Please use a larger batch size. 
2. I would suggest to just try all the blocks.

**Comment by dongdk** [comment]
is it suitable for sdxl?

**Comment by raywang4** [comment]
We have not tested it on Stable Diffusion, but Dispersive Loss can in theory be applied on any generative model, including SD-XL.

**Comment by supremePSS** [comment]
Why is disp_loss only applied to the last feature in the act list in the code?

disp_loss = 0
if "return_act" in model_kwargs and model_kwargs['return_act']:
    model_output, act = model_output
    disp_loss = self.disp_loss(act[-1])  （the act[-1] means the last feature in the act list, right?）

**Comment by raywang4** [comment]
We used a slightly different setting in PyTorch than the original JAX implementation. See "Differences from JAX" section.

**Comment by dongdk** [comment]
> We have not tested it on Stable Diffusion, but Dispersive Loss can in theory be applied on any generative model, including SD-XL.

which layer output do u suggest for sdxl?

**Comment by raywang4** [comment]
Again, I would suggest to try all layers.

**Comment by dongdk** [comment]
> Again, I would suggest to try all layers.

thanks, i will try it
